### PR TITLE
Job UCR Schema Design

### PR DESCRIPTION
**Pull request by kensipe**
Job UCR Schema Design

Summary:
Schema For UCR in Metronome

JIRA issues:  DCOS_OSS-4435

### CODE REVIEW

File: api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json [hunk 1]
```diff
@@ -79,6 +79,38 @@
...
+                "kind": {
+                  "type": "string",
+                  "documentation": "The type of image",
+                  "enum": ["docker", "appc"],
+                  "default": "docker"
+                }
+              },
+              "required": ["id"]
+            },
+            "forcePullImage": {
```
Comment (timcharper) [inline api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json]: I think this should be in image

```
"image": {
  "id": "...",
  "kind": "docker",
  "forcePull": false
}

Comment (timcharper) [reply api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json]: https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/podContainer.raml#L32

File: api/src/main/resources/public/api/v1/schema/jobspec.schema.json [hunk 2]
```diff
@@ -80,6 +80,38 @@
...
+                "kind": {
+                  "type": "string",
+                  "documentation": "The type of image",
+                  "enum": ["docker", "appc"],
+                  "default": "docker"
+                }
+              },
+              "required": ["id"]
+            },
+            "forcePullImage": {
```
Comment (timcharper) [inline api/src/main/resources/public/api/v1/schema/jobspec.schema.json]: ditto with forcePullImage here

File: api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json (hunk 1 above)
Comment (kensipe) [reply api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json]: good catch... I was going mainly off what was put in the doc... but clearly that was incorrect.

File: api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json [hunk 3]
```diff
@@ -79,6 +79,38 @@
...
+                "kind": {
+                  "type": "string",
+                  "documentation": "The type of image",
+                  "enum": ["docker", "appc"],
+                  "default": "docker"
+                }
+              },
+              "required": ["id"]
+            },
+            "forcePull": {
```
Comment (timcharper) [inline api/src/main/resources/public/api/v0/schema/jobspec_v0.schema.json]: let's put this inside of the image object

File: api/src/main/resources/public/api/v1/schema/jobspec.schema.json [hunk 4]
```diff
@@ -80,6 +80,38 @@
...
+                "kind": {
+                  "type": "string",
+                  "documentation": "The type of image",
+                  "enum": ["docker", "appc"],
+                  "default": "docker"
+                }
+              },
+              "required": ["id"]
+            },
+            "forcePull": {
```
Comment (timcharper) [inline api/src/main/resources/public/api/v1/schema/jobspec.schema.json]: let's put this inside of image

**Pull request merged**